addon/components/polaris-banner.js

```diff
@@ -96,109 +79,117 @@ export default class PolarisBanner extends Component {
 
-  @computed('title')
   get headingId() {
-    if (isBlank(this.title)) {
+    if (isBlank(this.args.title)) {
       return null;
     }
 
-    return `${guidFor(this)}-heading`;
+    return `${guidFor(this)}Heading`;
   }
 
-  @computed('status')
   get statusClass() {
-    let { status } = this;
+    let { status } = this.args;
     if (isBlank(status) || !supportedStatuses.includes(status)) {
       return null;
     }
 
     return `Polaris-Banner--status${capitalize(status)}`;
   }
 
-  @computed('statusClass', 'hasDismiss', 'withinContentContainer', 'class')
   get cssClasses() {
     let cssClasses = ['Polaris-Banner'];
     if (this.statusClass) {
       cssClasses.push(this.statusClass);
     }
-    if (this.hasDismiss) {
+    if (this.args.onDismiss) {
       cssClasses.push('Polaris-Banner--hasDismiss');
     }
+
+    if (this.shouldShowFocus) {
+      cssClasses.push('Polaris-Banner--keyFocused');
+    }
+
     if (this.withinContentContainer) {
       cssClasses.push('Polaris-Banner--withinContentContainer');
     } else {
       cssClasses.push('Polaris-Banner--withinPage');
     }
-    if (this.class) {
-      cssClasses.push(this.class);
+
+    if (this.polaris.features.newDesignLanguage) {
+      cssClasses.push('Polaris-Banner--newDesignLanguage');
     }
 
     return cssClasses.join(' ');
   }
 
-  init() {
-    super.init(...arguments);
-
-    deprecate(
-      `[PolarisBanner] Passing 'action' is deprecated! Please use 'primaryAction' instead`,
-      !this.action,
-      {
-        id: 'ember-polaris.polaris-banner.action-arg',
-        until: '7.0.0',
-      }
-    );
+  get useBannerAttributes() {
+    let { newDesignLanguage } = this.polaris.features;
+
+    switch (this.args.status) {
+      case 'success':
+        return {
+          defaultIcon: 'CircleTickMajor',
+          iconColor: newDesignLanguage ? 'success' : 'greenDark',
+          ariaRoleType: 'status',
+        };
+
+      case 'info':
+        return {
+          defaultIcon: 'CircleInformationMajor',
+          iconColor: newDesignLanguage ? 'highlight' : 'tealDark',
+          ariaRoleType: 'status',
+        };
+
+      case 'warning':
+        return {
+          defaultIcon: 'CircleAlertMajor',
+          iconColor: newDesignLanguage ? 'warning' : 'yellowDark',
+          ariaRoleType: 'alert',
+        };
+
+      case 'critical':
+        return {
+          defaultIcon: 'CircleDisabledMajor',
+          iconColor: newDesignLanguage ? 'critical' : 'redDark',
+          ariaRoleType: 'alert',
+        };
+
+      default:
+        return {
+          defaultIcon: newDesignLanguage
+            ? 'CircleInformationMajor'
+            : 'FlagMajor',
+          iconColor: newDesignLanguage ? 'base' : 'inkLighter',
+          ariaRoleType: 'status',
+        };
+    }
+  }
+
+  @action
+  handleBlur(e) {
+    e.currentTarget.blur();
+    this.shouldShowFocus = false;
+  }
+
+  @action
+  handleKeyUp() {
+    this.shouldShowFocus = true;
```

Comment:
not really sure if this should be a problem, but react implementation explicitly checks that the focus event is triggered on this component's main HTML element and wondering if we need to do that too _(talking about [this](https://github.com/Shopify/polaris-react/blob/9a167d3565573739b4f2dfa8e516bc8a13a0f179/src/components/Banner/Banner.tsx#L261))_

Reply:
Ah, yeah not sure how needed it is but I added it anyway.